src/common/signal_handling.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include <csignal>
+
+// SIGNAL HANDLING
+
+// The Marian signal handlers set global flags that thread can
```

Comment:
This is a generic and general mechanism so that people who write executables based on Marian as a library can easily implement custom behavior in response to signals received. The explanation of what it is "actually used for" is immediately below in the rest of the comment.